diff --git a/scripts/cross-build.sh b/scripts/cross-build.sh
index d520d3a..d4fc7dd 100755
--- a/scripts/cross-build.sh
+++ b/scripts/cross-build.sh
@@ -140,3 +140,3 @@ build() {
 
-    if [ "$ALIAS" = "win32" ]; then
+    if [ "$ALIAS" = "win32" ] || [ "$ALIAS" = "win64" ]; then
         COMPONENTS=2
@@ -188,2 +188,5 @@ case ${BUILD_TARGET} in
         ;;
+    win64|x86_64-mingw)
+        build x86_64-w64-mingw32 "win64"
+        ;;
     *)
